src/libappimage/desktop_integration/integrator/Integrator.cpp

```diff
@@ -270,7 +270,15 @@ namespace appimage {
                 }
 
                 void setExecutionPermission() {
-                    bf::permissions(appImage.getPath(), bf::owner_read | bf::owner_exe | bf::add_perms);
+                    if (access(appImage.getPath().c_str(), X_OK) != F_OK)
```

Comment:
Please stick with boost:
https://stackoverflow.com/a/9776289

Reply:
That was my first attempt, but it's not sufficient to check if the user has execution permissions, have to check if the user is the owner, also the group can have execution permissions then you have to check if the user belongs to the group and finally you can check if if the others have execution permissions.


`access` does all of that already and comes from `unistd.h` which is a header that we already depend on, therefore I think is better to reuse than to remake the wheel.